src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -118,7 +118,7 @@ internal Repository(bool needDir, IExceptionContext ectx)
 
         private static string GetShortTempDir()
         {
-            var rnd = RandomUtils.Create();
+            var rnd = RandomUtils.Create(Guid.NewGuid().GetHashCode());
             string path;
```

Comment:
Is there an associated bug? Are we seeing an issue here?

Reply:
yes, this PR is to fix below issue:
[https://github.com/dotnet/machinelearning/issues/5210](https://github.com/dotnet/machinelearning/issues/5210)

---
In reply to: [438920139](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438920139) [](ancestors = 438920139)

Reply:
❔  Why not just use `Path.GetRandomFileName()`?

Reply:
The entire implementation of `GetShortTempDir()` can be reduced to this:

```
var path = Path.Combine(Path.GetTempPath(), "ml_tests", Path.GetRandomFileName());
Directory.CreateDirectory(path);
return path;
```

The **ml_tests** subdirectory allows a user to easily identify and delete stray folders created by tests. You definitely don't want to place the folders directly in `%TEMP%` which is what happens today.

Reply:
Yes, please use `Path.GetRandomFileName()`. The original code had `RandomUtils.Create`, but we have seen it cause conflicts when it comes to file names. We have started replacing those with `Path.GetRandomFileName()`

---
In reply to: [438940202](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438940202) [](ancestors = 438940202)

Reply:
If you have a repro, please add a test to make sure that this fixed.

---
In reply to: [438936814](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438936814) [](ancestors = 438936814,438920139)

Reply:
No, I don't have repro, user who reported this issue also don't have repro on his local env, he get error from their product environment but not able to provide a repro

---
In reply to: [438942892](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438942892) [](ancestors = 438942892,438936814,438920139)

Reply:
I see that `Path.GetRandomFileName()`is indeed superior at generating file and/or directory names, so I'm in favor of using this too. It does seem to be [limited ](https://thedeveloperblog.com/random-string)to generating just 11 characters, though this will probably be more than enough.

It also seems interesting that here `Path.GetTempPath()` has been added, but `Path.GetRandomFileName()` wasn't`. I wonder if there's a reason why `Path.GetRandomFileName()` wasn't used in the first place.

Thanks @sharwell and @harishsk for the heads up.

Reply:
Thanks, that make sense

---
In reply to: [438942324](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438942324) [](ancestors = 438942324,438940202)

Reply:
Yeah, that is great idea, one little modification is I want to change "ml_tests" to "ml_dotnet" as this is not for test usage.

---
In reply to: [438941863](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438941863) [](ancestors = 438941863)

Reply:
checked the implementation again, Path.GetRandomFileName() is already added at line 239 by Jon Wood at 1/31/2020

---
In reply to: [438971881](https://github.com/dotnet/machinelearning/pull/5229#discussion_r438971881) [](ancestors = 438971881,438941863)